Automattic/wp-calypso
# Earn: Convert controller to typescript

## Proposed Changes

**You can see the specific changes in this commit**: https://github.com/Automattic/wp-calypso/pull/81465/commits/bb80e53d10927f25462582c24159730b9222d0fc

Refactors the Earn controller to TypeScript. Because of the file name change from .js to .ts, Github is not seeing these as the same file. That can make it harder to review specific changes. So I've pushed the commits such that the file rename and the actual changes are separate, and the specific changes are visible in the second commit.



## Testing Instructions



Checkout this branch, go to `http://calypso.localhost:3000/earn/YOURDOMAIN`, and click around different tabs and sections to confirm navigation still works the same as before. You'll want to connect Stripe if you haven't so that you can navigate between sections behind the Connect to Stripe screen.
